### PR TITLE
initial Firebase auth hookup

### PR DESCRIPTION
**Pull request by kenyee**
Starting this off since we're going to need it eventually.
Only does Google login but easy enough to add Twitter login, etc. once we set up API keys.
Also doesn't persist login state which it would need to do eventually.
And if you need to test this, it has to be done using a release build or your SHA1 dev key has to be added to the Firebase project.

### CODE REVIEW

**Comment by CCorrado** [comment]
I linked the appropriate tags, since I feel this is strongly related to #131

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/MainActivity.kt [hunk 1]
```diff
@@ -264,10 +291,52 @@ class MainActivity : AppCompatActivity() {
...
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == RC_SIGN_IN) {
+            val response = IdpResponse.fromResultIntent(data)
+
+            if (resultCode == Activity.RESULT_OK) {
+                user = FirebaseAuth.getInstance().currentUser
+                navView.menu.findItem(R.id.nav_login_logout).title = getString(R.string.str_logout)
```
[the remark this answers is not in the document]
Comment (kenyee) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/MainActivity.kt]: I'm not sure we need to.  This is purely login/signup via Firebase Auth.  Once you're logged in, we can present a "Profile" section for a user to fill out (which we can jump to in the onActivityResult if we detect it's their first login).

**Comment by CCorrado** [comment]
Looks good @kenyee !! I am planning on getting to the user profile stuff tonight.

**Comment by kenyee** [comment]
ok..merging so you have it in master

**Pull request merged**